creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -364,39 +361,48 @@ class SnapbackSM {
    * @returns {string} the endpoint of the newly selected secondary. Should not be a node in the
    *        current replica set.
    */
-  async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet }) {
+  async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet }) {
     // Fetch all the healthy nodes while disabling sync checks to select nodes for new replica set
-    if (!this.healthyNodes) {
-      const { services: healthyServicesMap } = await this.audiusLibs.ServiceProvider.autoSelectCreatorNodes({
-        performSyncCheck: false
-      })
+    // Note: sync checks are disabled because there should not be any syncs occurring for a particular user
+    // on a new replica set. Also, the sync check logic is coupled with a user state on the userStateManager.
+    // There will be an explicit clock value check on the newly selected replica set nodes instead.
+    const { services: healthyServicesMap } = await this.audiusLibs.ServiceProvider.autoSelectCreatorNodes({
```

Comment:
why fetch the entire healthyServices list from chain every on every single `issueUpdateReplicaSetOp`()? wouldn't it make more sense to fetch this outside the loop in the outer most function and pass it in, and just filter from it for each reconfig op? this slows things down quite a bit

Reply:
the below logic of 373 can also be moved outside with this code